accord-core/src/main/java/accord/local/Command.java

```diff
@@ -1322,8 +1421,9 @@ static Command.PreAccepted preaccept(Command command, CommonAttributes attrs, Ti
         }
         else if (command.status() == Status.AcceptedInvalidate && command.executeAt() == null)
         {
+            // TODO (now): reconsider this special-casing
             Command.Accepted accepted = command.asAccepted();
-            return Command.Accepted.accepted(attrs, accepted.saveStatus(), executeAt, ballot, accepted.accepted());
+            return Command.Accepted.accepted(attrs, SaveStatus.enrich(accepted.saveStatus(), SaveStatus.PreAccepted.known), executeAt, ballot, accepted.accepted());
```

Comment:
Because we told somebody we had recorded the definition; if we now lose it (or appear to lose it), and the coordinator relies on our having recorded it we may lose a quorum and be unable to reconstruct a transaction that had been agreed.